File: src/types/startup.ts
export type StartupSource =
  | "RegistryRun"
  | "RegistryRunOnce"
  | "StartupFolder"
  | "TaskScheduler";

export type StartupScope = "CurrentUser" | "AllUsers";

export type StartupImpact = "High" | "Medium" | "Low" | "NotMeasured";

export interface StartupEntry {
  id: string;
  name: string;
  command: string;
  source: StartupSource;
  scope: StartupScope;
  enabled: boolean;
  publisher: string | null;
  impact: StartupImpact;
}

File: src-tauri/src/models/startup.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct StartupEntry {
    pub id: String,
    pub name: String,
    pub command: String,
    pub source: StartupSource,
    pub scope: StartupScope,
    pub enabled: bool,
    pub publisher: Option<String>,
    pub impact: StartupImpact,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum StartupSource {
    RegistryRun,
    RegistryRunOnce,
    StartupFolder,
    TaskScheduler,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum StartupScope {
    CurrentUser,
    AllUsers,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum StartupImpact {
    High,
    Medium,
    Low,
    NotMeasured,
}
